Branch-SDK/src/io/branch/referral/ServerRequestQueue.java

```diff
@@ -106,7 +104,7 @@ public void run() {
         if (jsonStr != null) {
             try {
                 JSONArray jsonArr = new JSONArray(jsonStr);
-                for (int i = 0; i < Math.min(jsonArr.length(), MAX_ITEMS); i++) {
+                for (int i = 0, size = Math.min(jsonArr.length(), MAX_ITEMS); i < size; i++) {
```

Comment:
@dmkessler While this is ideal for iteration in a multi threaded condition, Is there a real chance for this to happen. The JSON array is created by a string retrieved from shared preference. Shared preference is thread safe. Is there a chance in your application using multiple process like a remote service which can access the preference simultaneously. (just thinking of way to reproduce the issue)

Reply:
Because jsonArr is initialized with jsonStr, and because java is pass by value rather than pass by reference, we can get the size of the array once when we start the loop rather than each iteration which it was doing previously.  It's just a trivial optimization.  If you are unsure about it this does not affect the fix in this PR and can be left out.